### ISSUE TITLE
How to ensure dependencies are created or updated before workloads

### ISSUE THREAD

**Issue by CharlesQQ** [karmada-io/karmada#1817]
**Please provide an in-depth description of the question you have**:
If both the dependency and the workload image are created or updated, should the workload be synchronized to the member cluster before the dependency is successful?

**What do you think about this question?**:
If workload synced successfully but dependency failed, this might be a problem?
**Environment**:
- Karmada version:
- Kubernetes version:
- Others:

**Comment by RainbowMango** [comment]
Do you mean does the `workload` will be waiting for all the `dependencies` ready before going to member clusters?

**Comment by CharlesQQ** [comment]
Yes, if u don't do this, there may be problems with the new pod running; from what I've looked at the `dependencies distributor` controller's code, it doesn't seem to guarantee this

**Comment by XiShanYongYe-Chang** [comment]
This feature should solve this problem: https://github.com/karmada-io/karmada/blob/master/docs/CHANGELOG/CHANGELOG-1.11.md#cluster-level-resource-propagation-pause-and-resume

Please feel free to reopen the issue if you have the same problem.
/close
